src/USSTree.ts

```diff
@@ -276,33 +274,54 @@ export class USSTree implements vscode.TreeDataProvider<ZoweUSSNode> {
-                if (!remotepath) {
-                    vscode.window.showInformationMessage(localize("enterPattern.pattern", "No selection made."));
-                    return;
+            if (this.mHistory.getHistory().length > 0) {
+
+                const createPick = new utils.FilterDescriptor(USSTree.defaultDialogText);
+                const items: vscode.QuickPickItem[] = this.mHistory.getHistory().map((element) => new utils.FilterItem(element));
+                if (extension.ISTHEIA) {
+                    const options1: vscode.QuickPickOptions = {
+                        placeHolder: localize("searchHistory.options.prompt", "Select a filter")
+                    };
+                    // get user selection
+                    const choice = (await vscode.window.showQuickPick([createPick, ...items], options1));
+                    if (!choice) {
+                        vscode.window.showInformationMessage(localize("enterPattern.pattern", "No selection made."));
+                        return;
+                    }
+                    remotepath = choice === createPick ? "" : choice.label;
+                } else {
+                    const quickpick = vscode.window.createQuickPick();
+                    quickpick.placeholder = localize("searchHistory.options.prompt", "Select a filter");
+                    quickpick.items = [createPick, ...items];
+                    quickpick.ignoreFocusOut = true;
+                    quickpick.show();
+                    const choice = await utils.resolveQuickPickHelper(quickpick);
+                    quickpick.hide();
+                    if (!choice) {
+                        vscode.window.showInformationMessage(localize("enterPattern.pattern", "No selection made."));
+                        return;
+                    }
+                    if (choice instanceof utils.FilterDescriptor) {
+                        if (quickpick.value) {
+                            remotepath = quickpick.value;
+                        }
+                    } else {
+                        remotepath = choice.label;
+                    }
                 }
             }
-            if (remotepath === USSTree.defaultDialogText) {
-                // manually entering a search
+            if (!remotepath) {
+                // manually entering a search - switch to an input box
                 const options: vscode.InputBoxOptions = {
                     prompt: localize("ussFilterPrompt.option.prompt.search",
-                        "Search Unix System Services (USS) by entering a path name starting with a /"),
+                        "Create a new filter"),
                     value: sessionNode.fullPath
                 };
                 // get user input
                 remotepath = await vscode.window.showInputBox(options);
-                if (!remotepath) {
+                if (!remotepath || remotepath.length === 0) {
```

Comment:
If I remember correctly empty strings evaluate to `falsy`, so don't know if the second part is even reachable.
No change required!

Reply:
You are correct. Ironically I found that out that whilst creating these tests but obviously didn't apply it here